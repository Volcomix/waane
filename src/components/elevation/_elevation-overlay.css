:root,
:host {
  --elevation1-overlay-opacity: var(--w-elevation1-overlay-opacity, 0.05);
  --elevation2-overlay-opacity: var(--w-elevation2-overlay-opacity, 0.07);
  --elevation3-overlay-opacity: var(--w-elevation3-overlay-opacity, 0.08);
  --elevation4-overlay-opacity: var(--w-elevation4-overlay-opacity, 0.09);
  --elevation6-overlay-opacity: var(--w-elevation6-overlay-opacity, 0.11);
  --elevation8-overlay-opacity: var(--w-elevation8-overlay-opacity, 0.12);
  --elevation12-overlay-opacity: var(--w-elevation12-overlay-opacity, 0.14);
  --elevation16-overlay-opacity: var(--w-elevation16-overlay-opacity, 0.15);
  --elevation24-overlay-opacity: var(--w-elevation24-overlay-opacity, 0.16);

  --elevation1-overlay: linear-gradient(
    rgba(var(--on-surface), var(--elevation1-overlay-opacity)),
    rgba(var(--on-surface), var(--elevation1-overlay-opacity))
  );

  --elevation2-overlay: linear-gradient(
    rgba(var(--on-surface), var(--elevation2-overlay-opacity)),
    rgba(var(--on-surface), var(--elevation2-overlay-opacity))
  );

  --elevation3-overlay: linear-gradient(
    rgba(var(--on-surface), var(--elevation3-overlay-opacity)),
    rgba(var(--on-surface), var(--elevation3-overlay-opacity))
  );

  --elevation4-overlay: linear-gradient(
    rgba(var(--on-surface), var(--elevation4-overlay-opacity)),
    rgba(var(--on-surface), var(--elevation4-overlay-opacity))
  );

  --elevation6-overlay: linear-gradient(
    rgba(var(--on-surface), var(--elevation6-overlay-opacity)),
    rgba(var(--on-surface), var(--elevation6-overlay-opacity))
  );

  --elevation8-overlay: linear-gradient(
    rgba(var(--on-surface), var(--elevation8-overlay-opacity)),
    rgba(var(--on-surface), var(--elevation8-overlay-opacity))
  );

  --elevation12-overlay: linear-gradient(
    rgba(var(--on-surface), var(--elevation12-overlay-opacity)),
    rgba(var(--on-surface), var(--elevation12-overlay-opacity))
  );

  --elevation16-overlay: linear-gradient(
    rgba(var(--on-surface), var(--elevation16-overlay-opacity)),
    rgba(var(--on-surface), var(--elevation16-overlay-opacity))
  );

  --elevation24-overlay: linear-gradient(
    rgba(var(--on-surface), var(--elevation24-overlay-opacity)),
    rgba(var(--on-surface), var(--elevation24-overlay-opacity))
  );
}
