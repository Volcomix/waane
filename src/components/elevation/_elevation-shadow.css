:root,
:host {
  --elevation-shadow-color: var(--w-elevation-shadow-color, 0, 0, 0);
  --elevation-umbra-opacity: var(--w-elevation-umbra-opacity, 0.2);
  --elevation-penumbra-opacity: var(--w-elevation-penumbra-opacity, 0.14);
  --elevation-ambient-opacity: var(--w-elevation-ambient-opacity, 0.12);

  --elevation1-umbra: var(--w-elevation1-umbra, 0px 2px 1px -1px);
  --elevation2-umbra: var(--w-elevation2-umbra, 0px 3px 1px -2px);
  --elevation3-umbra: var(--w-elevation3-umbra, 0px 3px 3px -2px);
  --elevation4-umbra: var(--w-elevation4-umbra, 0px 2px 4px -1px);
  --elevation6-umbra: var(--w-elevation6-umbra, 0px 3px 5px -1px);
  --elevation8-umbra: var(--w-elevation8-umbra, 0px 5px 5px -3px);
  --elevation12-umbra: var(--w-elevation12-umbra, 0px 7px 8px -4px);
  --elevation16-umbra: var(--w-elevation16-umbra, 0px 8px 10px -5px);
  --elevation24-umbra: var(--w-elevation24-umbra, 0px 11px 15px -7px);

  --elevation1-penumbra: var(--w-elevation1-penumbra, 0px 1px 1px 0px);
  --elevation2-penumbra: var(--w-elevation2-penumbra, 0px 2px 2px 0px);
  --elevation3-penumbra: var(--w-elevation3-penumbra, 0px 3px 4px 0px);
  --elevation4-penumbra: var(--w-elevation4-penumbra, 0px 4px 5px 0px);
  --elevation6-penumbra: var(--w-elevation6-penumbra, 0px 6px 10px 0px);
  --elevation8-penumbra: var(--w-elevation8-penumbra, 0px 8px 10px 1px);
  --elevation12-penumbra: var(--w-elevation12-penumbra, 0px 12px 17px 2px);
  --elevation16-penumbra: var(--w-elevation16-penumbra, 0px 16px 24px 2px);
  --elevation24-penumbra: var(--w-elevation24-penumbra, 0px 24px 38px 3px);

  --elevation0-ambient: var(--w-elevation0-ambient, 0px 0px 0px 0px);
  --elevation1-ambient: var(--w-elevation1-ambient, 0px 1px 3px 0px);
  --elevation2-ambient: var(--w-elevation2-ambient, 0px 1px 5px 0px);
  --elevation3-ambient: var(--w-elevation3-ambient, 0px 1px 8px 0px);
  --elevation4-ambient: var(--w-elevation4-ambient, 0px 1px 10px 0px);
  --elevation6-ambient: var(--w-elevation6-ambient, 0px 1px 18px 0px);
  --elevation8-ambient: var(--w-elevation8-ambient, 0px 3px 14px 2px);
  --elevation12-ambient: var(--w-elevation12-ambient, 0px 5px 22px 4px);
  --elevation16-ambient: var(--w-elevation16-ambient, 0px 6px 30px 5px);
  --elevation24-ambient: var(--w-elevation24-ambient, 0px 9px 46px 8px);

  --elevation1-shadow: var(--elevation1-umbra)
      rgba(var(--elevation-shadow-color), var(--elevation-umbra-opacity)),
    var(--elevation1-penumbra)
      rgba(var(--elevation-shadow-color), var(--elevation-penumbra-opacity)),
    var(--elevation1-ambient)
      rgba(var(--elevation-shadow-color), var(--elevation-ambient-opacity));

  --elevation2-shadow: var(--elevation2-umbra)
      rgba(var(--elevation-shadow-color), var(--elevation-umbra-opacity)),
    var(--elevation2-penumbra)
      rgba(var(--elevation-shadow-color), var(--elevation-penumbra-opacity)),
    var(--elevation2-ambient)
      rgba(var(--elevation-shadow-color), var(--elevation-ambient-opacity));

  --elevation3-shadow: var(--elevation3-umbra)
      rgba(var(--elevation-shadow-color), var(--elevation-umbra-opacity)),
    var(--elevation3-penumbra)
      rgba(var(--elevation-shadow-color), var(--elevation-penumbra-opacity)),
    var(--elevation3-ambient)
      rgba(var(--elevation-shadow-color), var(--elevation-ambient-opacity));

  --elevation4-shadow: var(--elevation4-umbra)
      rgba(var(--elevation-shadow-color), var(--elevation-umbra-opacity)),
    var(--elevation4-penumbra)
      rgba(var(--elevation-shadow-color), var(--elevation-penumbra-opacity)),
    var(--elevation4-ambient)
      rgba(var(--elevation-shadow-color), var(--elevation-ambient-opacity));

  --elevation6-shadow: var(--elevation6-umbra)
      rgba(var(--elevation-shadow-color), var(--elevation-umbra-opacity)),
    var(--elevation6-penumbra)
      rgba(var(--elevation-shadow-color), var(--elevation-penumbra-opacity)),
    var(--elevation6-ambient)
      rgba(var(--elevation-shadow-color), var(--elevation-ambient-opacity));

  --elevation8-shadow: var(--elevation8-umbra)
      rgba(var(--elevation-shadow-color), var(--elevation-umbra-opacity)),
    var(--elevation8-penumbra)
      rgba(var(--elevation-shadow-color), var(--elevation-penumbra-opacity)),
    var(--elevation8-ambient)
      rgba(var(--elevation-shadow-color), var(--elevation-ambient-opacity));

  --elevation12-shadow: var(--elevation12-umbra)
      rgba(var(--elevation-shadow-color), var(--elevation-umbra-opacity)),
    var(--elevation12-penumbra)
      rgba(var(--elevation-shadow-color), var(--elevation-penumbra-opacity)),
    var(--elevation12-ambient)
      rgba(var(--elevation-shadow-color), var(--elevation-ambient-opacity));

  --elevation16-shadow: var(--elevation16-umbra)
      rgba(var(--elevation-shadow-color), var(--elevation-umbra-opacity)),
    var(--elevation16-penumbra)
      rgba(var(--elevation-shadow-color), var(--elevation-penumbra-opacity)),
    var(--elevation16-ambient)
      rgba(var(--elevation-shadow-color), var(--elevation-ambient-opacity));

  --elevation24-shadow: var(--elevation24-umbra)
      rgba(var(--elevation-shadow-color), var(--elevation-umbra-opacity)),
    var(--elevation24-penumbra)
      rgba(var(--elevation-shadow-color), var(--elevation-penumbra-opacity)),
    var(--elevation24-ambient)
      rgba(var(--elevation-shadow-color), var(--elevation-ambient-opacity));
}
