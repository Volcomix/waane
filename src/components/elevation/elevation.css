.elevation--z1 {
  box-shadow: var(--elevation-shadow, var(--elevation1-shadow));
  background-image: var(--elevation-overlay, var(--elevation1-overlay));
}

.elevation--z2 {
  box-shadow: var(--elevation-shadow, var(--elevation2-shadow));
  background-image: var(--elevation-overlay, var(--elevation2-overlay));
}

.elevation--z3 {
  box-shadow: var(--elevation-shadow, var(--elevation3-shadow));
  background-image: var(--elevation-overlay, var(--elevation3-overlay));
}

.elevation--z4 {
  box-shadow: var(--elevation-shadow, var(--elevation4-shadow));
  background-image: var(--elevation-overlay, var(--elevation4-overlay));
}

.elevation--z6 {
  box-shadow: var(--elevation-shadow, var(--elevation6-shadow));
  background-image: var(--elevation-overlay, var(--elevation6-overlay));
}

.elevation--z8 {
  box-shadow: var(--elevation-shadow, var(--elevation8-shadow));
  background-image: var(--elevation-overlay, var(--elevation8-overlay));
}

.elevation--z12 {
  box-shadow: var(--elevation-shadow, var(--elevation12-shadow));
  background-image: var(--elevation--overlay, var(--elevation12-overlay));
}

.elevation--z16 {
  box-shadow: var(--elevation-shadow, var(--elevation16-shadow));
  background-image: var(--elevation--overlay, var(--elevation16-overlay));
}

.elevation--z24 {
  box-shadow: var(--elevation-shadow, var(--elevation24-shadow));
  background-image: var(--elevation--overlay, var(--elevation24-overlay));
}
