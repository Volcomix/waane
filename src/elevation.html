<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <title>Elevation</title>
    <meta name="description" content="Elegation test page" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style>
      @import './components/variables.css';
      @import './components/elevation/elevation.css';

      .theme {
        display: flex;
        flex-wrap: wrap;
        justify-content: space-between;
        padding: 8px;
        background-color: rgb(var(--background));
      }

      .theme--light {
        --background: 255, 255, 255;
        --surface: 255, 255, 255;
        --on-surface: 0, 0, 0;
        --elevation-overlay: none;
      }

      .surface {
        display: flex;
        justify-content: center;
        align-items: center;
        margin: 16px;
        min-width: 200px;
        min-height: 100px;
        background-color: rgb(var(--surface));
        color: rgba(var(--on-surface), var(--high-emphasis));
      }
    </style>
  </head>
  <body>
    <div class="theme theme--light">
      <div class="surface elevation--z1">1dp</div>
      <div class="surface elevation--z2">2dp</div>
      <div class="surface elevation--z3">3dp</div>
      <div class="surface elevation--z4">4dp</div>
      <div class="surface elevation--z6">6dp</div>
      <div class="surface elevation--z8">8dp</div>
      <div class="surface elevation--z12">12dp</div>
      <div class="surface elevation--z16">16dp</div>
      <div class="surface elevation--z24">24dp</div>
    </div>
    <div class="theme">
      <div class="surface elevation--z1">1dp</div>
      <div class="surface elevation--z2">2dp</div>
      <div class="surface elevation--z3">3dp</div>
      <div class="surface elevation--z4">4dp</div>
      <div class="surface elevation--z6">6dp</div>
      <div class="surface elevation--z8">8dp</div>
      <div class="surface elevation--z12">12dp</div>
      <div class="surface elevation--z16">16dp</div>
      <div class="surface elevation--z24">24dp</div>
    </div>
  </body>
</html>
