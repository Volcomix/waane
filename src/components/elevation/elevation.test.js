let moduleHandle

beforeAll(async () => {
  await page.goto('http://localhost:8080/test.html')
  moduleHandle = await page.evaluateHandle(/* js */ `
    import('./components/elevation/elevation.js')
  `)
})

for (let z = 0; z < 25; z++) {
  it(`returns elevation styles for z=${z}`, async () => {
    const elevation = await page.evaluate(
      ({ default: elevation }, z) => elevation(z),
      moduleHandle,
      z,
    )
    expect(elevation).toMatchSnapshot()
  })
}
