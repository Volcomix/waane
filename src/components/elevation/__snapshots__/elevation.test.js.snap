// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns elevation styles for z=0 1`] = `
"
    z-index: 0;
    box-shadow: var(--shadow, 0px 0px 0px 0px rgba(var(--shadow-color), var(--umbra-opacity)), 0px 0px 0px 0px rgba(var(--shadow-color), var(--penumbra-opacity)), 0px 0px 0px 0px rgba(var(--shadow-color), var(--ambient-opacity)));
    background-image: var(--overlay, linear-gradient(rgba(var(--on-surface), 0), rgba(var(--on-surface), 0)));
  "
`;

exports[`returns elevation styles for z=1 1`] = `
"
    z-index: 1;
    box-shadow: var(--shadow, 0px 2px 1px -1px rgba(var(--shadow-color), var(--umbra-opacity)), 0px 1px 1px 0px rgba(var(--shadow-color), var(--penumbra-opacity)), 0px 1px 3px 0px rgba(var(--shadow-color), var(--ambient-opacity)));
    background-image: var(--overlay, linear-gradient(rgba(var(--on-surface), 0.05), rgba(var(--on-surface), 0.05)));
  "
`;

exports[`returns elevation styles for z=2 1`] = `
"
    z-index: 2;
    box-shadow: var(--shadow, 0px 3px 1px -2px rgba(var(--shadow-color), var(--umbra-opacity)), 0px 2px 2px 0px rgba(var(--shadow-color), var(--penumbra-opacity)), 0px 1px 5px 0px rgba(var(--shadow-color), var(--ambient-opacity)));
    background-image: var(--overlay, linear-gradient(rgba(var(--on-surface), 0.07), rgba(var(--on-surface), 0.07)));
  "
`;

exports[`returns elevation styles for z=3 1`] = `
"
    z-index: 3;
    box-shadow: var(--shadow, 0px 3px 3px -2px rgba(var(--shadow-color), var(--umbra-opacity)), 0px 3px 4px 0px rgba(var(--shadow-color), var(--penumbra-opacity)), 0px 1px 8px 0px rgba(var(--shadow-color), var(--ambient-opacity)));
    background-image: var(--overlay, linear-gradient(rgba(var(--on-surface), 0.08), rgba(var(--on-surface), 0.08)));
  "
`;

exports[`returns elevation styles for z=4 1`] = `
"
    z-index: 4;
    box-shadow: var(--shadow, 0px 2px 4px -1px rgba(var(--shadow-color), var(--umbra-opacity)), 0px 4px 5px 0px rgba(var(--shadow-color), var(--penumbra-opacity)), 0px 1px 10px 0px rgba(var(--shadow-color), var(--ambient-opacity)));
    background-image: var(--overlay, linear-gradient(rgba(var(--on-surface), 0.09), rgba(var(--on-surface), 0.09)));
  "
`;

exports[`returns elevation styles for z=5 1`] = `
"
    z-index: 5;
    box-shadow: var(--shadow, 0px 3px 5px -1px rgba(var(--shadow-color), var(--umbra-opacity)), 0px 5px 8px 0px rgba(var(--shadow-color), var(--penumbra-opacity)), 0px 1px 14px 0px rgba(var(--shadow-color), var(--ambient-opacity)));
    background-image: var(--overlay, linear-gradient(rgba(var(--on-surface), 0.1), rgba(var(--on-surface), 0.1)));
  "
`;

exports[`returns elevation styles for z=6 1`] = `
"
    z-index: 6;
    box-shadow: var(--shadow, 0px 3px 5px -1px rgba(var(--shadow-color), var(--umbra-opacity)), 0px 6px 10px 0px rgba(var(--shadow-color), var(--penumbra-opacity)), 0px 1px 18px 0px rgba(var(--shadow-color), var(--ambient-opacity)));
    background-image: var(--overlay, linear-gradient(rgba(var(--on-surface), 0.11), rgba(var(--on-surface), 0.11)));
  "
`;

exports[`returns elevation styles for z=7 1`] = `
"
    z-index: 7;
    box-shadow: var(--shadow, 0px 4px 5px -2px rgba(var(--shadow-color), var(--umbra-opacity)), 0px 7px 10px 1px rgba(var(--shadow-color), var(--penumbra-opacity)), 0px 2px 16px 1px rgba(var(--shadow-color), var(--ambient-opacity)));
    background-image: var(--overlay, linear-gradient(rgba(var(--on-surface), 0.11), rgba(var(--on-surface), 0.11)));
  "
`;

exports[`returns elevation styles for z=8 1`] = `
"
    z-index: 8;
    box-shadow: var(--shadow, 0px 5px 5px -3px rgba(var(--shadow-color), var(--umbra-opacity)), 0px 8px 10px 1px rgba(var(--shadow-color), var(--penumbra-opacity)), 0px 3px 14px 2px rgba(var(--shadow-color), var(--ambient-opacity)));
    background-image: var(--overlay, linear-gradient(rgba(var(--on-surface), 0.12), rgba(var(--on-surface), 0.12)));
  "
`;

exports[`returns elevation styles for z=9 1`] = `
"
    z-index: 9;
    box-shadow: var(--shadow, 0px 5px 6px -3px rgba(var(--shadow-color), var(--umbra-opacity)), 0px 9px 12px 1px rgba(var(--shadow-color), var(--penumbra-opacity)), 0px 3px 16px 2px rgba(var(--shadow-color), var(--ambient-opacity)));
    background-image: var(--overlay, linear-gradient(rgba(var(--on-surface), 0.12), rgba(var(--on-surface), 0.12)));
  "
`;

exports[`returns elevation styles for z=10 1`] = `
"
    z-index: 10;
    box-shadow: var(--shadow, 0px 6px 6px -3px rgba(var(--shadow-color), var(--umbra-opacity)), 0px 10px 14px 1px rgba(var(--shadow-color), var(--penumbra-opacity)), 0px 4px 18px 3px rgba(var(--shadow-color), var(--ambient-opacity)));
    background-image: var(--overlay, linear-gradient(rgba(var(--on-surface), 0.13), rgba(var(--on-surface), 0.13)));
  "
`;

exports[`returns elevation styles for z=11 1`] = `
"
    z-index: 11;
    box-shadow: var(--shadow, 0px 6px 7px -4px rgba(var(--shadow-color), var(--umbra-opacity)), 0px 11px 15px 1px rgba(var(--shadow-color), var(--penumbra-opacity)), 0px 4px 20px 3px rgba(var(--shadow-color), var(--ambient-opacity)));
    background-image: var(--overlay, linear-gradient(rgba(var(--on-surface), 0.13), rgba(var(--on-surface), 0.13)));
  "
`;

exports[`returns elevation styles for z=12 1`] = `
"
    z-index: 12;
    box-shadow: var(--shadow, 0px 7px 8px -4px rgba(var(--shadow-color), var(--umbra-opacity)), 0px 12px 17px 2px rgba(var(--shadow-color), var(--penumbra-opacity)), 0px 5px 22px 4px rgba(var(--shadow-color), var(--ambient-opacity)));
    background-image: var(--overlay, linear-gradient(rgba(var(--on-surface), 0.14), rgba(var(--on-surface), 0.14)));
  "
`;

exports[`returns elevation styles for z=13 1`] = `
"
    z-index: 13;
    box-shadow: var(--shadow, 0px 7px 8px -4px rgba(var(--shadow-color), var(--umbra-opacity)), 0px 13px 19px 2px rgba(var(--shadow-color), var(--penumbra-opacity)), 0px 5px 24px 4px rgba(var(--shadow-color), var(--ambient-opacity)));
    background-image: var(--overlay, linear-gradient(rgba(var(--on-surface), 0.14), rgba(var(--on-surface), 0.14)));
  "
`;

exports[`returns elevation styles for z=14 1`] = `
"
    z-index: 14;
    box-shadow: var(--shadow, 0px 7px 9px -4px rgba(var(--shadow-color), var(--umbra-opacity)), 0px 14px 21px 2px rgba(var(--shadow-color), var(--penumbra-opacity)), 0px 5px 26px 4px rgba(var(--shadow-color), var(--ambient-opacity)));
    background-image: var(--overlay, linear-gradient(rgba(var(--on-surface), 0.14), rgba(var(--on-surface), 0.14)));
  "
`;

exports[`returns elevation styles for z=15 1`] = `
"
    z-index: 15;
    box-shadow: var(--shadow, 0px 8px 9px -5px rgba(var(--shadow-color), var(--umbra-opacity)), 0px 15px 22px 2px rgba(var(--shadow-color), var(--penumbra-opacity)), 0px 6px 28px 5px rgba(var(--shadow-color), var(--ambient-opacity)));
    background-image: var(--overlay, linear-gradient(rgba(var(--on-surface), 0.14), rgba(var(--on-surface), 0.14)));
  "
`;

exports[`returns elevation styles for z=16 1`] = `
"
    z-index: 16;
    box-shadow: var(--shadow, 0px 8px 10px -5px rgba(var(--shadow-color), var(--umbra-opacity)), 0px 16px 24px 2px rgba(var(--shadow-color), var(--penumbra-opacity)), 0px 6px 30px 5px rgba(var(--shadow-color), var(--ambient-opacity)));
    background-image: var(--overlay, linear-gradient(rgba(var(--on-surface), 0.15), rgba(var(--on-surface), 0.15)));
  "
`;

exports[`returns elevation styles for z=17 1`] = `
"
    z-index: 17;
    box-shadow: var(--shadow, 0px 8px 11px -5px rgba(var(--shadow-color), var(--umbra-opacity)), 0px 17px 26px 2px rgba(var(--shadow-color), var(--penumbra-opacity)), 0px 6px 32px 5px rgba(var(--shadow-color), var(--ambient-opacity)));
    background-image: var(--overlay, linear-gradient(rgba(var(--on-surface), 0.15), rgba(var(--on-surface), 0.15)));
  "
`;

exports[`returns elevation styles for z=18 1`] = `
"
    z-index: 18;
    box-shadow: var(--shadow, 0px 9px 11px -5px rgba(var(--shadow-color), var(--umbra-opacity)), 0px 18px 28px 2px rgba(var(--shadow-color), var(--penumbra-opacity)), 0px 7px 34px 6px rgba(var(--shadow-color), var(--ambient-opacity)));
    background-image: var(--overlay, linear-gradient(rgba(var(--on-surface), 0.15), rgba(var(--on-surface), 0.15)));
  "
`;

exports[`returns elevation styles for z=19 1`] = `
"
    z-index: 19;
    box-shadow: var(--shadow, 0px 9px 12px -6px rgba(var(--shadow-color), var(--umbra-opacity)), 0px 19px 29px 2px rgba(var(--shadow-color), var(--penumbra-opacity)), 0px 7px 36px 6px rgba(var(--shadow-color), var(--ambient-opacity)));
    background-image: var(--overlay, linear-gradient(rgba(var(--on-surface), 0.15), rgba(var(--on-surface), 0.15)));
  "
`;

exports[`returns elevation styles for z=20 1`] = `
"
    z-index: 20;
    box-shadow: var(--shadow, 0px 10px 13px -6px rgba(var(--shadow-color), var(--umbra-opacity)), 0px 20px 31px 3px rgba(var(--shadow-color), var(--penumbra-opacity)), 0px 8px 38px 7px rgba(var(--shadow-color), var(--ambient-opacity)));
    background-image: var(--overlay, linear-gradient(rgba(var(--on-surface), 0.16), rgba(var(--on-surface), 0.16)));
  "
`;

exports[`returns elevation styles for z=21 1`] = `
"
    z-index: 21;
    box-shadow: var(--shadow, 0px 10px 13px -6px rgba(var(--shadow-color), var(--umbra-opacity)), 0px 21px 33px 3px rgba(var(--shadow-color), var(--penumbra-opacity)), 0px 8px 40px 7px rgba(var(--shadow-color), var(--ambient-opacity)));
    background-image: var(--overlay, linear-gradient(rgba(var(--on-surface), 0.16), rgba(var(--on-surface), 0.16)));
  "
`;

exports[`returns elevation styles for z=22 1`] = `
"
    z-index: 22;
    box-shadow: var(--shadow, 0px 10px 14px -6px rgba(var(--shadow-color), var(--umbra-opacity)), 0px 22px 35px 3px rgba(var(--shadow-color), var(--penumbra-opacity)), 0px 8px 42px 7px rgba(var(--shadow-color), var(--ambient-opacity)));
    background-image: var(--overlay, linear-gradient(rgba(var(--on-surface), 0.16), rgba(var(--on-surface), 0.16)));
  "
`;

exports[`returns elevation styles for z=23 1`] = `
"
    z-index: 23;
    box-shadow: var(--shadow, 0px 11px 14px -7px rgba(var(--shadow-color), var(--umbra-opacity)), 0px 23px 36px 3px rgba(var(--shadow-color), var(--penumbra-opacity)), 0px 9px 44px 8px rgba(var(--shadow-color), var(--ambient-opacity)));
    background-image: var(--overlay, linear-gradient(rgba(var(--on-surface), 0.16), rgba(var(--on-surface), 0.16)));
  "
`;

exports[`returns elevation styles for z=24 1`] = `
"
    z-index: 24;
    box-shadow: var(--shadow, 0px 11px 15px -7px rgba(var(--shadow-color), var(--umbra-opacity)), 0px 24px 38px 3px rgba(var(--shadow-color), var(--penumbra-opacity)), 0px 9px 46px 8px rgba(var(--shadow-color), var(--ambient-opacity)));
    background-image: var(--overlay, linear-gradient(rgba(var(--on-surface), 0.16), rgba(var(--on-surface), 0.16)));
  "
`;
